src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.BIO.Custom.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+internal static partial class Interop
+{
+    internal static class CustomBio
+    {
+        private static IntPtr _customBioMethodStructure;
+        private static CreateDelegate _createDelegate;
+        private static DestroyDelegate _destroyDelegate;
+        private unsafe static ControlDelegate _controlDelegate;
+        private unsafe static ReadDelegate _readDelegate;
+        private unsafe static WriteDelegate _writeDelegate;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate int CreateDelegate(bio_st* bio);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate int ReadDelegate(IntPtr bio, void* buf, int size);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate int WriteDelegate(IntPtr bio, void* buf, int num);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate long ControlDelegate(IntPtr bio, Crypto.BIO_CTRL cmd, long num, void* ptr);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int DestroyDelegate(IntPtr bio);
+
+        static CustomBio()
+        {
+            CustomBio.Initialize();
+        }
+
+        internal unsafe static void Initialize()
+        {
+            _createDelegate = Create;
+            _controlDelegate = Control;
+            _destroyDelegate = Destroy;
+            _writeDelegate = Write;
+            _readDelegate = Read;
+
+            var name = Marshal.StringToHGlobalAnsi("Managed Bio");
+            var bioStruct = new bio_method_st()
+            {
+                create = _createDelegate,
+                name = name,
+                type = BIO_TYPE.BIO_TYPE_SOURCE_SINK,
+                destroy = _destroyDelegate,
+                ctrl = _controlDelegate,
+                bread = _readDelegate,
+                bwrite = _writeDelegate,
+            };
+
+            var memory = Marshal.AllocHGlobal(Marshal.SizeOf<bio_method_st>());
```

Comment:
removed many many var violations (suddenly the analyzers are working in my VS :) )